src/parallel/executor.cpp

```diff
@@ -442,6 +442,32 @@ void Executor::RescheduleTask(shared_ptr<Task> &task_p) {
 	}
 }
 
+bool Executor::ResultCollectorIsBlocked() const {
+	if (completed_pipelines + 1 != total_pipelines) {
+		// The result collector is always in the last pipeline
+		return false;
+	}
+	if (to_be_rescheduled_tasks.empty()) {
```

Comment:
Not sure, but could this be racy in some situations? Like when a task hasn't been added to `to_be_rescheduled_tasks`? I guess for now we always hold the executor lock when calling this so its fine?

Reply:
Hmm fair, it's a private method though so I would argue it's expected the lock is held when this is called